### ISSUE TITLE
Thread safety

### ISSUE THREAD

**Issue by sebglazebrook** [gchp/rustbox#41]
Hi,

I've been experimenting with rustbox and wanted to use it to across threads.

According to this commit it looks like I can: https://github.com/gchp/rustbox/pull/8

But then I get this error when I try to use it in a thread:

```
the trait `core::marker::Send` is not implemented for the type `*mut ()`
```

Which I think is coming from here:

https://github.com/gchp/rustbox/blob/7e76a04e7c78a96e00b0435ef5065ce2381c7492/src/rustbox.rs#L221

Which indicated that termbox isn't thread safe. 

Can rustbox be used in threads?

If so, any tips on how? I've tried wrapping it in `Arc::new(Mutex::new(Rustbox::init())` but without luck.

Cheers

**Comment by sebglazebrook** [comment]
Retouching this one to show I'm still interested. :-)

**Comment by SirVer** [comment]
+1 about this issue.

**Comment by gchp** [comment]
Sorry for the delay getting back to you, I was on vacation without my laptop :)

The short answer is, I'm not sure. Which I realise is not a very satisfying answer! The whole thread-safety side of things is a bit of a gray area to me at the moment unfortunately. I'll do some digging and see if I can figure it out!

**Comment by sebglazebrook** [comment]
I've experimented using it across threads by disabling the PhantomData (https://github.com/sebglazebrook/rustbox) and while it compiled it did hang during execution.

Haven't had a chance to investigate whether it's my code or something else.

Will let you know if I find anything.

**Comment by sebglazebrook** [comment]
To get back to you about the above. The issue I was seeing was due to my code. 

Now that I've fixed that I've found rustbox to be fine across threads aonce wrapped in an Arc and Mutex.

I've created a PR (https://github.com/gchp/rustbox/pull/44) to remove PhantomData unless there is another reason why it should be there.

**Comment by jmacdonald** [comment]
@gchp I think we can safely close this, as it's addressed by #81.
